This is loosely based on upstream patch
c23abafbdcdd3754bea50f1452fed36da5515379 from iproute2 package.

There were some other changes from RHEL6 kernel include/linux/if_link.h
file that were added so that the offset in the enum would be correct.
There may be other changes needed, but this is the minimum.

This patch depends on kernel bug 845081.

diff --git a/include/linux/if_link.h b/include/linux/if_link.h
index 15692ae..c39662e 100644
--- a/include/linux/if_link.h
+++ b/include/linux/if_link.h
@@ -80,6 +80,10 @@ enum {
 	IFLA_IFALIAS,
 	IFLA_NUM_VF,		/* Number of VFs if device is SR-IOV PF */
 	IFLA_VFINFO_LIST,
+	IFLA_STATS64,
+	IFLA_VF_PORTS,
+	IFLA_PORT_SELF,
+	IFLA_EXT_MASK,          /* Extended info mask, VFs, etc */
 	__IFLA_MAX
 };
 
diff --git a/include/linux/rtnetlink.h b/include/linux/rtnetlink.h
index bb56966..a059785 100644
--- a/include/linux/rtnetlink.h
+++ b/include/linux/rtnetlink.h
@@ -597,6 +597,9 @@ struct tcamsg {
 #define TCA_ACT_TAB 1 /* attr type must be >=1 */	
 #define TCAA_MAX 1
 
+/* New extended info filters for IFLA_EXT_MASK */
+#define RTEXT_FILTER_VF		(1 << 0)
+
 /* End of information exported to user level */
 
 
